test/shared/src/test/scala/zio/test/TestAspectSpec.scala

```diff
@@ -75,70 +75,64 @@ object TestAspectSpec extends DefaultRuntime {
+        exceptionAssertion[NullPointerException]
       )
       succeeded(spec)
     }
 
   def failureDoesNotMakeTestsPassOnUnexpectedException: Future[Boolean] =
     unsafeRunToFuture {
-      val spec = test("failureDoesNotMakeTestsPassOnUnexpectedException")(
+      val spec = test("test")(
         assert(throw new NullPointerException(), isFalse)
-      ) @@ failure[
-        Nothing
-      ](failsWithException[IllegalArgumentException])
+      ) @@ failure(exceptionAssertion[IllegalArgumentException])
       failed(spec)
     }
 
   def failureDoesNotMakeTestsPassOnUnexpectedCause: Future[Boolean] =
     unsafeRunToFuture {
-      val spec = test("failureDoesNotMakeTestsPassOnUnexpectedCause")(assert(throw new RuntimeException(), isFalse)) @@ failure[
-        String
-      ](
-        isCase("Runtime", { case TestFailure.Runtime(e) => Some(e); case _ => None }, equalTo(Cause.fail("boom")))
+      val spec = test("test")(assert(throw new RuntimeException(), isFalse)) @@ failure(
+        isCase[TestFailure[String], Cause[String]]("Runtime", {
+          case TestFailure.Runtime(e) => Some(e); case _ => None
+        }, equalTo(Cause.fail("boom")))
       )
       failed(spec)
     }
 
   def failureMakesTestsPassOnAnyAssertionFailure: Future[Boolean] =
     unsafeRunToFuture {
-      val spec = test("failureMakesTestsPassOnAnyAssertionFailure")(assert(true, equalTo(false))) @@ failure
+      val spec = test("test")(assert(true, equalTo(false))) @@ failure
       succeeded(spec)
     }
 
   def failureMakesTestsPassOnExpectedAssertionFailure: Future[Boolean] =
     unsafeRunToFuture {
-      val spec = test("failureMakesTestsPassOnExpectedAssertionFailure")(assert(true, equalTo(false))) @@ failure[
-        Nothing
-      ] {
-        isCase("Assertion", { case TestFailure.Assertion(result) => Some(result); case _ => None }, anything)
-      }
+      val spec = test("test")(assert(true, equalTo(false))) @@ failure(
+        isCase[TestFailure[Any], TestResult]("Assertion", {
+          case TestFailure.Assertion(result) => Some(result); case _ => None
+        }, anything)
+      )
       succeeded(spec)
     }
 
   def failureDoesNotMakesTestsPassOnUnexpectedAssertionFailure: Future[Boolean] =
     unsafeRunToFuture {
-      val spec = test("failureDoesNotMakesTestsPassOnUnexpectedAssertionFailure")(assert(true, equalTo(false))) @@ failure[
-        Nothing
-      ] {
-        isCase(
+      val spec = test("test")(assert(true, equalTo(false))) @@ failure(
+        isCase[TestFailure[Boolean], TestResult](
           "Assertion", { case TestFailure.Assertion(result) => Some(result); case _ => None },
           equalTo(assert(42, equalTo(42)))
         )
-      }
+      )
       failed(spec)
     }
 
-  private def failsWithException[T <: Throwable](implicit ct: ClassTag[T]): Assertion[Any] =
+  private def exceptionAssertion[T <: Throwable](implicit ct: ClassTag[T]): Assertion[TestFailure[Any]] =
```

Comment:
For what it's worth I thought your original name of `failsWithException` was actually clearer here.